primitives/arithmetic/src/biguint.rs

```diff
@@ -214,7 +214,7 @@ impl BigUint {
 
 				if let Some(v) = u.checked_sub(v) {
 					if let Some(v2) = v.checked_sub(k) {
-						t = v2 % B;
+						t = v2;
```

Comment:
this is good to me as `v2` is less than `u` which is less than `B`

Reply:
yes